lib/saluki-app/src/logging.rs

```diff
@@ -115,19 +133,47 @@ fn initialize_logging_inner(
             .replace(LoggingAPIHandler::new(shared_level_filter.clone(), reload_handle));
     }
 
+    let adp_log_file = std::env::var("DD_ADP_LOG_FILE").unwrap_or(DEFAULT_ADP_LOG_FILE.to_string());
+    let log_file_max_size = std::env::var("DD_LOG_FILE_MAX_SIZE")
+        .map(|s| s.parse::<u64>().unwrap_or(DEFAULT_LOG_FILE_MAX_SIZE))
+        .unwrap_or(DEFAULT_LOG_FILE_MAX_SIZE);
+    let log_file_max_rolls = std::env::var("DD_LOG_FILE_MAX_ROLLS")
+        .map(|s| s.parse::<usize>().unwrap_or(DEFAULT_LOG_FILE_MAX_ROLLS))
+        .unwrap_or(DEFAULT_LOG_FILE_MAX_ROLLS);
+    let file_appender = BasicRollingFileAppender::new(
+        adp_log_file,
+        RollingConditionBasic::new().max_size(log_file_max_size),
+        log_file_max_rolls,
+    )?;
+    let (file_nb, guard) = tracing_appender::non_blocking(file_appender);
+    let file_level_filter = EnvFilter::builder()
+        .with_default_directive(default_level.unwrap_or(LevelFilter::INFO).into())
+        .with_env_var("DD_LOG_LEVEL")
+        .from_env_lossy();
+
     if is_json {
         let json_layer = initialize_tracing_json();
         tracing_subscriber::registry()
             .with(json_layer.with_filter(filter_layer))
+            .with(
+                tracing_subscriber::fmt::Layer::new()
+                    .with_writer(file_nb)
+                    .with_filter(file_level_filter),
```

Comment:
This is going to use the default logging format of `tracing-subscriber` which isn't going to match either format we use here for stdout.

It's not clear to me if setting the logging to JSON mode in the Core Agent (or subagents) would also affect their log files... but I know that we would at least want the non-JSON mode to have the same formatting for the log file.

Might be able to just clone the JSON/pretty layers, or might need to initialize 'em twice, not sure.